internal/e2e/conditions.go

```diff
@@ -32,6 +32,21 @@ func (c *ConditionExtension) ResourceMatch(obj k8s.Object, matchFetcher func(obj
 	}
 }
 
+func (c *ConditionExtension) PodSucceeded(pod k8s.Object) apimachinerywait.ConditionWithContextFunc {
+	return func(ctx context.Context) (done bool, err error) {
+		if err := c.resources.Get(ctx, pod.GetName(), pod.GetNamespace(), pod); err != nil {
+			return false, err
+		}
+		status := pod.(*v1.Pod).Status
+		if status.Phase == v1.PodSucceeded {
+			done = true
```

Comment:
this is a bit awkward given that all other returns are full, can this be
```suggestion
			return true, nil
```